Access control: allow hiding roles (#46358)
grafana/grafana: docs/sources/http_api/access_control.md

| version     | number              | Yes      | Version of the role. Must be incremented for update to work.                                                                                                                   |
| name        | string              | Yes      | Name of the role.                                                                                                                                                              |
| description | string              | No       | Description of the role.                                                                                                                                                       |
| displayName | string              | No       | Display name of the role, visible in the UI.                                                                                                                                   |
| group       | string              | No       | The group name the role belongs to.                                                                                                                                            |

> | hidden      | boolean             | No       | A flag indicating if the role is hidden or not. If set to `true`, role will not appear in the role picker and will not be listed by API endpoints unless explicitly specified. |

Reviewer: ```suggestion
| hidden      | boolean             | No       | Specify whether the role is hidden or not. If set to `true`, then the role does not show in the role picker. It will not be listed by API endpoints unless explicitly specified. |
```

> | permissions | List of Permissions | No       | The full list of permissions the role should have after the update.                                                                                                            |

Reviewer: ```suggestion
| permissions | List of Permissions | No       | The full list of permissions for the role after the update.                                                                                                            |
```

Lists the roles that have been directly assigned to a given user. The list does not include built-in roles (Viewer, Editor, Admin or Grafana Admin), and it does not include roles that have been inherited from a team.

Query Parameters:

> - `includeHidden`: boolean. Optional. Include roles that are set to `hidden`.

Reviewer: ```suggestion
- `includeHidden`: Optional. Set to `true` to include roles that are `hidden`.
```

#### JSON body schema

| Field Name    | Date Type | Required | Description                                                                                                                                          |
| ------------- | --------- | -------- | ---------------------------------------------------------------------------------------------------------------------------------------------------- |

> | global        | boolean   | No       | A flag indicating if the assignment is global or not. If set to `false`, the default org ID of the authenticated user will be used from the request. |

Reviewer: ```suggestion
| global        | boolean   | No       | Specify whether the assignment is global or not. If set to `false`, then the default org ID of the authenticated user is used from the request. |
```

| roleUids      | list      | Yes      | List of role UIDs.                                                                                                                                   |

> | includeHidden | boolean   | No       | A flag indicating if hidden roles should be updated by the request.                                                                                  |

Reviewer: I am sorry @gamab, I updated your suggestion by error. How do I revert?

> | includeHidden | boolean   | No       | A flag indicating if hidden roles should be updated by the request.                                                                                  |

Reviewer 2: Don't worry about it :)

Gets all built-in role assignments.

Query Parameters:

> - `includeHidden`: boolean. Optional. Include roles that are set to `hidden`.

Reviewer: ```suggestion
- `includeHidden`: Optional. Set to `true` to include roles that are `hidden`.
```
